### ISSUE TITLE
SDK Target Platforms .targets/.props files not loading on MAUI Project

### ISSUE THREAD

**Issue by stmoor** [dotnet/maui#5847]
## Description
The following .props/.targets files don't always load on a MAUI project. 
- Microsoft.Android.Sdk.SupportedPlatforms.targets
- Microsoft.iOS.Sdk.SupportedTargetPlatforms.props
- Microsoft.MacCatalyst.Sdk.SupportedTargetPlatforms.props

This happens even when the `TargetFrameworks` property contains all the target platforms (net6.0-ios, net6.0-android, and net6.0-maccatalyst). In fact, it only loads the file associated with the platform that is the first one in the list of values for the `TargetFrameworks` property. The windows file seems to always be loaded regardless of whether or not net6.0-windows is a target platform. 

## Request
Can we always load those files regardless of whether or not the related platform is a target listed in the `TargetFrameworks` property? 

This is necessary because we use those files to read the versions supported to display them to the user for them to select in a combo box (see picture below). So, we always need to display those values even when the framework is not selected, therefore those files should always be loaded. 

![image](https://user-images.githubusercontent.com/77985069/161838650-095cd367-7ffe-496c-8ff4-a4b7e9f6fcd7.png)

### Steps to Reproduce

1. Create a MAUI project
2. Make sure net6.0-android, net6.0-ios and net6.0-maccatalyst is listed in the `TagetFrameworks` property. 
3. Click the "Show all files" icon in the Solution Explorer
4. Search for "supported" 
5. Note that only the `Microsoft.NET.WindowsSdkSupportedTargetPlatforms.props ` and the file related to the first platform listed in the `TargetFrameworks` is loaded. 
6. Switch the order of the target platforms in the `TargetFrameworks` property. 
7. Unload and reload the project
8. Repeat steps 3-5

See picture below for an example:
![image](https://user-images.githubusercontent.com/77985069/161839671-bfb122e7-dc38-450e-b59f-88309586ad88.png)

### Version with bug

Preview 14 (current)

### Last version that worked well

Preview 14 (current)

### Affected platforms

iOS, Android, macOS

### Affected platform versions

n/a

**Comment by stmoor** [comment]
@japarson

**Comment by Redth** [comment]
@rolfbjarne @dalexsoto @jonathanpeppers  @jonpryor @mattleibow 

I'm not entirely sure how this would be accomplished.  I suppose MAUI would need to import these?

**Comment by rolfbjarne** [comment]
Do you look at the `SdkSupportedTargetPlatformVersion` item group or the platform-specific item group (such as `iOSSdkSupportedTargetPlatformVersion`) to get the supported versions?

A better version might be to add another item group, loaded from AutoImports.props (so every build gets it) where everybody add their versions and where the sdk is added as metadata. Something like this:

```xml
	<ItemGroup>
		<SupportedTargetPlatformVersionForSdk Include="15.2" Sdk="iOS" />
		<SupportedTargetPlatformVersionForSdk Include="15.4" Sdk="iOS" />
        ...
	</ItemGroup>
```

**Comment by stmoor** [comment]
> Do you look at the `SdkSupportedTargetPlatformVersion` item group or the platform-specific item group (such as `iOSSdkSupportedTargetPlatformVersion`) to get the supported versions?
> 
> A better version might be to add another item group, loaded from AutoImports.props (so every build gets it) where everybody add their versions and where the sdk is added as metadata. Something like this:
> 
> ```
> 	<ItemGroup>
> 		<SupportedTargetPlatformVersionForSdk Include="15.2" Sdk="iOS" />
> 		<SupportedTargetPlatformVersionForSdk Include="15.4" Sdk="iOS" />
>         ...
> 	</ItemGroup>
> ```

Yes I do. It seems like those files are separately generated by each project, like there is a script that generates the iOS file for example. So not sure how you would consolidate them that way.

**Comment by mattleibow** [comment]
Where are you currently getting the Windows TFMs from? Should we be doing something like that?

**Comment by stmoor** [comment]
I get the Windows TFMs from this file `Microsoft.NET.WindowsSdkSupportedTargetPlatforms.props`. It is always loaded. I noticed that there are no conditions on loading this file, however, for the other three, there are similar conditions like the one below: 

![image](https://user-images.githubusercontent.com/77985069/162042427-8d605595-42ce-43db-b314-726020d758ec.png)

This condition is in the `WorkloadsManifest.targets` file. You can see the chain of files in the picture below: 

![image](https://user-images.githubusercontent.com/77985069/162042764-5ceb230d-425e-4d59-b68e-06855d712823.png)

**Comment by dsplaisted** [comment]
There aren't any conditions on the Windows .props file because it's not in a separate workload (though it may be someday).

Is it possible to evaluate the platform-specific evaluation instances to get the supported platform versions?  IE to get the iOS supported platform versions, look at the evaluation for the project with the iOS TargetFramework.  That should have the `SdkSupportedTargetPlatformVersion` items for iOS, and similarly for Windows and Android when the TargetFramework indicates that platform.

If that's not possible, it may be possible to move some of this information into the workload AutoImport.targets, where it will always be part of evaluation if the workload is installed.  But it would need to be a different item than is currently being used, as the `SdkSupportedTargetPlatformVersion` item is used for available versions of the current target platform.

**Comment by tmeschter** [comment]
I believe the "Imports" node only shows you the imports for the active project configuration, which generally corresponds to the first item in `TargetFrameworks`. That doesn't mean the others aren't being loaded, just that you can't see them in Solution Explorer.

Since the Solution Explorer only shows us info for the active configuration, it is still possible that those imports aren't being pulled in to their correct configuration, or it may mean that the `IDynamicEnumValuesProvider` (somehow) ends up reading data only from the active configuration, rather than the correct configuration.

**Comment by Redth** [comment]
@stmoor or @tmeschter is there any further action here?

**Comment by Zhanglirong-Winnie** [comment]
This issue has been verified to be repro with Visual Studio Enterprise 17.7.0 Preview 5.0.
![Screenshot 2023-08-07 151646](https://github.com/dotnet/maui/assets/77434865/0d99b99a-ac76-4b2c-884f-68a44766b9a6)

**Comment by kubaflo** [comment]
Thanks for the report. The remaining problem belongs to Visual Studio, the debugger/editor, or the .NET MAUI VS Code extension rather than the MAUI runtime.

Please report it through the [Visual Studio feedback tool](https://learn.microsoft.com/visualstudio/ide/how-to-report-a-problem-with-visual-studio) or the [.NET MAUI VS Code extension repository](https://github.com/microsoft/vscode-dotnettools/issues), as appropriate. I am closing this MAUI issue so the runtime backlog only tracks work actionable in this repository; if you file the tooling report, please link it here.

**Comment by kubaflo** [comment]
Reopening — this issue should not have been closed.

It carries `s/verified` (a maintainer or partner reproduced it), `s/triaged` (it already went through triage), `partner/ui-tools`, `partner/winui` (a partner team is tracking it). Those labels mean the report was already confirmed as a real product defect, so it is not eligible for backlog cleanup closure. Closure here requires specific fix evidence (a merged PR that closes it, or a `fixed-in-*` label), and there was none.

Sorry for the noise, and thanks to everyone who flagged it.

> Disclosure: this comment was generated by AI.
